template/filter.c.erb

```diff
@@ -64,13 +66,26 @@ consume_upstream_messages:; /* Workaround §6.8.1 Labeled Statements */
       bt_message_iterator_next(message_iterator_private_data->message_iterator,
                                &upstream_messages, &upstream_message_count);
 
+
   /* We cannot have any message left in the downstream queue
    * Hence it's safe to pass any error messages downstream */
```

Comment:
Please edit this comment, it's not true anymore :) due to the finalization

Reply:
This assumption is TRUE, since `bt_message_iterator_next` is called when we do not have more messages in `common_data->queue`. Hence, in case of error, `btx_finalize_usr_data` is not called, so not messages are loaded to `common_data->queue`.

Reply:
We don't pass any error messages downstream anymore. In the case of the `STATUS_END` we don't pass it downstream the first time.